src/ui-test/tests/camel.settings.test.ts

```diff
@@ -0,0 +1,261 @@
+        const JMX_PROP = 'jmx:serverURL';
+
+        before(_setup());
+
+        after(function () {
+            resetUserSettings(CATALOG_PROVIDER_ID);
+        });
+
+        const PROVIDERS_LIST = [
+            // runtime provider, knative available, mongo available, jmx available
+            ['SPRINGBOOT', true, true, true],
+            ['QUARKUS', true, true, false],
+            ['KARAF', false, true, true]
+        ];
+
+        PROVIDERS_LIST.forEach(function (provider) {
+            const PROVIDER = provider.at(0).toString();
+            const KNATIVE_AV = Boolean(provider.at(1));
+            const MONGO_AV = Boolean(provider.at(2));
+            const JMX_AV = Boolean(provider.at(3));
+
+            describe(`${PROVIDER}`, function () {
+
+                before(async function () {
+                    await setRuntimeProvider(PROVIDER);
+                });
+
+                beforeEach(async function () {
+                    await VSBrowser.instance.openResources(path.join(RESOURCES, CAMEL_CONTEXT_XML));
+                    await waitUntilEditorIsOpened(driver, CAMEL_CONTEXT_XML);
+                });
+
+                afterEach(async function () {
+                    await closeEditor(CAMEL_CONTEXT_XML, false);
+                });
+
+                it('Knative component', async function () {
+                    await testComponent(KNATIVE, KNATIVE_PROP, KNATIVE_AV, 1);
+                });
+
+                it('Mongo component', async function () {
+                    await testComponent(MONGO, MONGO_PROP, MONGO_AV, 2);
+                });
+
+                it('JMX component', async function () {
+                    await testComponent(JMX, JMX_PROP, JMX_AV, 1);
+                });
+            });
+        });
+
+        /**
+         * Checks if component proposal is available. 
+         * 
+         * @param component Component to be tested.
+         * @param proposal Tested proposal for component.
+         * @param proposalAvailable Should be proposal available.
+         * @param proposalsCount Number of available proposals.
+         */
+        async function testComponent(component: string, proposal: string, proposalAvailable: boolean, proposalsCount: number): Promise<void> {
+            const editor = new TextEditor();
+            await editor.typeTextAt(3, XML_URI_POSITION, component);
+
+            contentAssist = await editor.toggleContentAssist(true) as ContentAssist;
+            const items = await contentAssist.getItems();
+
+            if (proposalAvailable) {
+                assert.equal(items.length, proposalsCount);
+                const proposalItem = await contentAssist.getItem(proposal);
+                assert.equal(await getTextExt(proposalItem), proposal);
+            } else {
+                assert.equal(items.length, 0);
+            }
+        }
+    });
+
+    describe('Camel catalog user preference version set', function () {
```

Comment:
because we are under `User preferences` header (describe) it would be sufficient to have label just simply `Camel Catalog version`